proof/refine/ARM/KHeap_R.thy

```diff
@@ -1473,6 +1475,25 @@ lemma get_sc_corres:
   apply (fastforce simp add: other_obj_relation_def)
 done
 
+lemma get_sc_corres_size:
+  "corres (\<lambda>sc sc'. sc_relation sc n sc' \<and> n + minSchedContextBits = objBits sc')
+     (sc_obj_at n ptr) (sc_at' ptr)
+     (get_sched_context ptr) (getSchedContext ptr)"
+  apply (rule corres_no_failI)
+   apply wp
+  apply (simp add: get_sched_context_def getSchedContext_def get_object_def
```

Comment:
Can some of these simps and clarsimps be combined? Or would it make more sense to leave them as is?

Reply:
Putting all together actually breaks the proof. It is possible that some of these are ok to combine, but some attempts I made didn’t work and I don’t think it is worth the effort to explore other possibilities right now. So, I’ll leave it like this for now.